tests/Helpers/JWKFetcherTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Auth0\Tests\Helpers;
+
+use Auth0\SDK\Helpers\JWKFetcher;
+use josegonzalez\Dotenv\Loader;
+use GuzzleHttp\Exception\RequestException;
+
+/**
+ * Class JWKFetcherTest
+ *
+ * @package Auth0\SDK\Helpers
+ */
+class JWKFetcherTest extends \PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Issuer for JWKS URL.
+     *
+     * @var string
+     */
+    public static $issuer;
+
+    /**
+     * Setup before class.
+     *
+     * @return void
+     */
+    public static function setUpBeforeClass()
+    {
+        $env_path = '.env';
+        if (file_exists($env_path)) {
+            $loader = new Loader($env_path);
+            $loader->parse()->putenv(true);
+        }
+
+        self::$issuer = 'https://'.getenv('DOMAIN').'/';
+        parent::setUpBeforeClass();
+    }
+
+    /**
+     * Test that a valid JWKS is fetched.
+     *
+     * @return void
+     */
+    public function testFetchJwks()
+    {
+        $fetcher = new JWKFetcher();
+        $keys    = $fetcher->fetchKeys( self::$issuer );
```

Comment:
why is this making a real request? Lets not depend on external stuff. Can you mock it please?

Reply:
All of the tests in this SDK are integration and hit a live endpoint. I have a backlog task to store JSON files to stop that but it's a big job.

Reply:
The issue is.. if you change your tenant settings and start using HS for example, all this tests will break. Ideally you will mock the request library to return a fixed response (e.g. success json or failure response). Then test that the JWKFetcher handles both scenarios fine.